File: src/main/scala/robosky/uplands/client/render/blockentity/ControlBlockEntityRenderer.scala
package robosky.uplands.client.render.blockentity

import com.mojang.blaze3d.platform.GlStateManager
import com.mojang.blaze3d.systems.RenderSystem

import net.minecraft.client.MinecraftClient
import net.minecraft.client.render.block.entity.{BlockEntityRenderDispatcher, BlockEntityRenderer}
import net.minecraft.client.render.{BufferBuilder, RenderLayer, Tessellator, VertexConsumer, VertexConsumerProvider, VertexFormats, WorldRenderer}
import net.minecraft.client.util.math.MatrixStack

import robosky.uplands.UplandsMod
import robosky.uplands.block.BlockRegistry
import robosky.uplands.block.bossroom.ControlBlockEntity

class ControlBlockEntityRenderer(dispatcher: BlockEntityRenderDispatcher)
    extends BlockEntityRenderer[ControlBlockEntity](dispatcher) {
  override def render(be: ControlBlockEntity, partialTicks: Float, matrix: MatrixStack, provider: VertexConsumerProvider, var5: Int, var6: Int): Unit = {
    val player = MinecraftClient.getInstance.player
    if ((player.isCreativeLevelTwoOp || player.isSpectator) && player.inventory.contains(UplandsMod.BOSSROOM_TECHNICAL_TAG)) {
      matrix.push()
      val minX: Double = be.bounds.minX
      val minY: Double = be.bounds.minY
      val minZ: Double = be.bounds.minZ
      val maxX: Double = be.bounds.maxX
      val maxY: Double = be.bounds.maxY
      val maxZ: Double = be.bounds.maxZ
      val consumer: VertexConsumer = provider.getBuffer(RenderLayer.getLines)
      WorldRenderer.drawBox(matrix, consumer, 0, 0, 0, 1, 1, 1, 1.0f, 1.0f, 0.0f, 1.0f)
      WorldRenderer.drawBox(matrix, consumer, minX, minY, minZ, maxX, maxY, maxZ, 0.0f, 0.0f, 0.0f, 1.0f)
      WorldRenderer.drawBox(matrix, consumer, minX, minY, minZ, maxX, maxY, maxZ, 1.0f, 1.0f, 1.0f, 1.0f)
      matrix.pop()
    }
  }
}

File: src/main/java/robosky/uplands/client/render/blockentity/ControlBlockEntityRenderer.java
package robosky.uplands.client.render.blockentity;

import robosky.uplands.block.bossroom.ControlBlockEntity;
import robosky.uplands.item.UplandsItemTags;

import net.minecraft.client.MinecraftClient;
import net.minecraft.client.render.RenderLayer;
import net.minecraft.client.render.VertexConsumer;
import net.minecraft.client.render.VertexConsumerProvider;
import net.minecraft.client.render.WorldRenderer;
import net.minecraft.client.render.block.entity.BlockEntityRenderDispatcher;
import net.minecraft.client.render.block.entity.BlockEntityRenderer;
import net.minecraft.client.util.math.MatrixStack;
import net.minecraft.entity.player.PlayerEntity;

public class ControlBlockEntityRenderer extends BlockEntityRenderer<ControlBlockEntity> {

    public ControlBlockEntityRenderer(BlockEntityRenderDispatcher dispatcher) {
        super(dispatcher);
    }

    @Override
    public void render(ControlBlockEntity be, float partialTicks, MatrixStack matrix, VertexConsumerProvider provider, int var5, int var6) {
        PlayerEntity player = MinecraftClient.getInstance().player;
        if (player != null && (player.isCreativeLevelTwoOp() || player.isSpectator()) && player.inventory.contains(UplandsItemTags.BOSSROOM_TECHNICAL)) {
            matrix.push();
            double minX = be.bounds().minX();
            double minY = be.bounds().minY();
            double minZ = be.bounds().minZ();
            double maxX = be.bounds().maxX();
            double maxY = be.bounds().maxY();
            double maxZ = be.bounds().maxZ();
            VertexConsumer consumer = provider.getBuffer(RenderLayer.getLines());
            WorldRenderer.drawBox(matrix, consumer, 0, 0, 0, 1, 1, 1, 1.0f, 1.0f, 0.0f, 1.0f);
            WorldRenderer.drawBox(matrix, consumer, minX, minY, minZ, maxX, maxY, maxZ, 0.0f, 0.0f, 0.0f, 1.0f);
            WorldRenderer.drawBox(matrix, consumer, minX, minY, minZ, maxX, maxY, maxZ, 1.0f, 1.0f, 1.0f, 1.0f);
            matrix.pop();
        }
    }
}
